src/decisionengine/framework/modules/SourceProxy.py

```diff
@@ -25,6 +25,7 @@
                         Parameter('retry_timeout', default=RETRY_TO, comment="Number of seconds to wait between retries."))
 class SourceProxy(Source.Source):
     def __init__(self, config):
+        super().__init__(config)
```

Comment:
You must remove line 37, otherwise `self.logger` is re-initialized unbound and the channel should be already bound
```
        self.logger = structlog.getLogger(CHANNELLOGGERNAME)
        self.logger = self.logger.bind(class_module=__name__.split(".")[-1], channel=self.channel_name)
```
to
```
        self.logger = self.logger.bind(class_module=__name__.split(".")[-1])
```

Reply:
Yes, you are right. I have that sitting in my repo but forgot to push it. Thanks. Doing that now.